PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -626,7 +626,17 @@ def miniAOD_customizeData(process):
     process.load("RecoPPS.Local.ctppsLocalTrackLiteProducer_cff")
     process.load("RecoPPS.ProtonReconstruction.ctppsProtons_cff")
     process.load("Geometry.VeryForwardGeometry.geometryRPFromDB_cfi")
+    process.load('L1Trigger.L1TGlobal.simGtExtFakeProd_cfi')
+    process.simGtExtFakeProd.bxFirst = cms.int32(-2)
+    process.simGtExtFakeProd.bxLast = cms.int32(2)
+    process.simGtExtFakeProd.setBptxAND = cms.bool(True)
+    process.simGtExtFakeProd.setBptxPlus = cms.bool(True)
+    process.simGtExtFakeProd.setBptxMinus = cms.bool(True)
+    process.simGtExtFakeProd.setBptxOR = cms.bool(True)
+    process.simGtExtFakeProd.tcdsRecordLabel = cms.InputTag("tcdsDigis","tcdsRecord")
     task = getPatAlgosToolsTask(process)
+    from Configuration.ProcessModifiers.run2_miniAOD_UL_cff import run2_miniAOD_UL
+    run2_miniAOD_UL.toModify(task, func=lambda t: t.add(process.simGtExtFakeProd))
```

Comment:
for the master branch I expect that this feature should be enabled by default.
So, `run2_miniAOD_UL.toModify` is not enough.
I see that a previous generic addition failed for some cases; is there a L1-related common modifier that would be expected to have `simGtExtFakeProd` working? (my somewhat wild guess is `(stage2L1Trigger_2018 & stage2L1Trigger_2021)`, just because the 2017 wf failed

Reply:
Hi @slava77 
To properly address this comment, I am trying to understand why `run2_miniAOD_UL.toModify` is not enough in the master branch ?  I guess you are saying that for 10_6_X it is fine to use this modifier ?

The way we currently designed the adding of the Unprefireable bit via this Producer is for reprocessing of AOD and re-production of miniAOD (UL), which is uniquely defined with `run2_miniAOD_UL` modifier. 

This producer (if configured with a non-empty `tcdsRecordLabel`) assumes that there is a TCDS record in the event, which is the case for the inputs of miniAOD_UL. So I would think that `run2_miniAOD_UL` modifier would be fine to use for controlling when this producer configured as such is to be added to the sequence .  @srimanob can comment further here for possible suggestions.

In any case, L1T era modifiers (for example `stage2L1Trigger_2018 & stage2L1Trigger_2021`) would not do the job.

Reply:
I'm lost, if I read your comment directly, a TCDS record is written only in 10_6_X data rereco campaign; that it is NOT created in any other standard workflows.

IIUC, it actually exists in the AOD or RECO outputs for a much larger set of workflows.

In the master version of the release, a new feature should be enabled by default in all (as much as possible) cases where it can work.
For the backport to 10_6_X the no-change policy for a production release requires that a new feature can be enabled only for `run2_miniAOD_UL`.